noobaa_sa/bucket.py

```diff
@@ -0,0 +1,121 @@
+from framework.connection import SSHConnection
+from noobaa_sa.defaults import MANAGE_NSFS
+import noobaa_sa.exceptions as e
+
+log = logging.getLogger(__name__)
+
+
+
+
+class BucketOperation:
+    """
+    Bucket operations
+    """
+    def __init__(self):
+        """
+        Initialize necessary variable
+        """
+        self.manage_nsfs = MANAGE_NSFS
+        self.config_root = config.ENV_DATA["config_root"]
+        self.conn = SSHConnection().connection
+
+    def createBucket(self, account_name, bucket_name, config_root=None):
+        """
+        Create bucket using CLI
+
+        Args:
+            config_root (str): Path to config root
+            account_name: User name
+            bucket_name: Name of the bucket
+        """
+        unwanted_log = "2>/dev/null"
+        base_cmd = f"sudo /usr/local/noobaa-core/bin/node {self.manage_nsfs} "
+        if config_root is None:
+            config_root = self.config_root
+        log.info("Gather user info before creating bucket")
+        cmd = f"{base_cmd}account status --config_root {config_root} --name {account_name} {unwanted_log}"
+        retcode, stdout, stderr = self.conn.exec_cmd(cmd)
+        log.info(retcode)
+        if retcode != 0:
+            raise e.AccountStatusFailed(
+                f"Failed to get status of account {stderr}"
+            )
+        account_info = re.findall(r'\{([^}]+)\}', stdout)
+        pattern = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
+        account_info = account_info[0].strip()
+        account_info = pattern.sub('', account_info)
+        account_info = account_info.replace("'","")
+        account_info_param = account_info.split(", ")
+        account_dict = {}
+        for item in account_info_param:
+            key, value = item.split(':')
+            account_dict[key.strip()] = value.strip()
+        account_email = str(account_dict['email'])
+        bucket_path = str(account_dict['new_buckets_path'])
+
+        cmd = f"{base_cmd} bucket add --config_root {config_root} --name {bucket_name} --email {account_email} --path {bucket_path} {unwanted_log}"
+        retcode, stdout, stderr = self.conn.exec_cmd(cmd)
+        if retcode != 0:
+            raise e.BucketCreationFailed(
+                f"Failed to create bucket {stderr}"
+            )
+        log.info("Bucket created successfully")
+
+
+    def listBucket(self, config_root=None):
+        """
+        Lists Buckets
+
+        Args:
+            config_root (str): Path to config root
+
+        """
+        if config_root is None:
+            config_root = self.config_root
+        log.info("Listing available buckets")
+        cmd = f"sudo /usr/local/noobaa-core/bin/node {self.manage_nsfs} bucket list --config_root {config_root}"
+        retcode, stdout, stderr = self.conn.exec_cmd(cmd)
+        if retcode != 0:
+            raise e.BucketListFailed(f"Listing of buckets failed with error {stderr}")
+        log.info(stdout)
```

Comment:
We should parse the output to return a list of bucket names / objects - it will be very useful for future testing.

Reply:
@sagihirshfeld As said above, this PR only checks commands are working correctly on latest build. I will send separate PR for extracting list of buckets from existing output in upcoming PRs